File: Ruby/season2/p021-026/p025.rb
# -*- coding: utf-8 -*-
=begin
　2017.5.2
　==と===
=end

puts "==","==="
puts "【(1..3) ? 2】"
p ((1..3)  == 2)
p ((1..3) === 2)
puts "【/exp/ ? \"expression\"】"
p (/exp/  == "expression")
p (/exp/ === "expression")
puts "【String ? \"学園\"】"
p (String  == "学園")
p (String === "学園")
puts "【\"学園\" ? String】"
p ("学園"  == String)
p ("学園" === String)
puts "【Array ? [1,1,7,1,1,7,\"学園\"]】"
p (Array  == [1,1,7,1,1,7,"学園"])
p (Array === [1,1,7,1,1,7,"学園"])

File: Python/season2/p024-025/p025.py
# coding: UTF-8
"""
　2018.4.20
　イテレータ関数
"""

def number_generator(x):
    if (x%2):
        return 3*x+1
    else:
        return x//2

def number_generator_generator():
  yield number_generator(1)
  yield number_generator(2)
  yield number_generator(3)
  yield number_generator(4)
  yield number_generator(5)
  
for num in number_generator_generator():
  print(num)